### PR TITLE
Use internal workspace-images to create and dog food dev environment images

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Use internal workspace full image `europe-docker.pkg.dev/gitpod-artifacts/docker-dev/workspace-base-images:full` for the creation of dev-environment image `eu.gcr.io/gitpod-core-dev/dev/dev-environment`.
This dev-environment image is produced by werft job. We want to consume this image in order to dog food our workspace images which are being generated by [dazzle-rewrite](https://github.com/gitpod-io/workspace-images/pull/593).
This will help us discover any issue with the base image quicker as gitpod developers would be actively using the dev env based on this image.

Update the werft build script to configure docker to use gcloud as credential helper in order to pull image from internal repo.

Fix werft secret creation of `gcp-registry-auth`. Using `echo` within the `exec` function of typescript resulted in conversion of all newline characters `\n` with actual newlines. i.e. something like below
```
echo -n "this\nis\ngitpod"
```
resulted in an output
```
this
is
gitpod
```
whereas we expected
```
this\nis\ngitpod
```
This broke the image pull secret.

This was not discovered until now as the image repo being used in the preview environment had public access. Introducing a private repo uncovered this issue.

## Current status
There's an issue with GPG keys for `llvm` and `docker` resulting in NO_PUBKEY errors, which was preventing install of `kubectl`. We've put a temporary hack in the dev image's Dockerfile to add the missing keys, and are working on solution for `workspace-images`. This way, users won't have a bad experience when using `add-apt-repository`.

Also, we're looking into an issue with the dev image's Dockerfile, it had some path trouble with Python utils (it didn't previously). This is an indicator that there are some path changes which are breaking.

These will not prevent us from dog fooding, but there is some more testing we have to do before sharing with Gitpodders.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/7042

## How to test

0. Test with the [Gitpod repo](https://prs-test-dazzle-rewrite.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod)
1. Checkout a new branch from HEAD of this branch and push it to repo.
2. This will run a werft build
2.1.  The build will use the internal workspace image -> Passing build means success
2.2. The build will create a dev-environment image which can be seen in the output tab of werft build -> This image is already updated in the .gitpod.yml of this PR (with a different tag). Opening this branch in gitpod -> Successful test
2.3. The build will deploy a preview env. -> Able to access gitpod dashboard -> Successful test
2.4. Use the gitpod installation in preview env to open a repo branch (eg. [this](https://github.com/princerachit/atheoscommune.github.io/blob/dev-env-test/.gitpod.yml)) which uses the internal workspace image. -> Workspace opens means successful test

[Open a workspace for the Gitpod repo using the dev image](https://prs-test-dazzle-rewrite.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod) Succeded ✅ 
[Helm installation](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.76/results) Succeded ✅ 
[Installer installation](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.77) Succeeded ✅ 
[Leeway No cache Build](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.91/results) Succeeded ✅ 

## Release Notes

```release-note
None
```

### PR Build Controls
- [ ] /werft with-helm=true 
- [x] /werft with-clean-slate-deployment=
- [ ] /werft with-vm=

### CODE REVIEW

**Comment by roboquat** [comment]
@princerachit: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **princerachit** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by kylos101** [comment]
@princerachit we should be able to remove the public key workarounds from the dev image now, I made a change to the `workspace-images` [here](https://github.com/gitpod-io/workspace-images/commit/d55ccabb9486a4883f93f04a89ed4634c169e0b0), and it works like a champ! :+1:  I tested on my local machine (recreating the issue before and after). :smile:

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **princerachit** after the PR has been reviewed.

Associated issue: *#7042*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by princerachit** [comment]
- [x] /werft with-helm=true 
- [x] /werft with-clean-slate-deployment=

**Comment by princerachit** [comment]
/werft run
- [x] /werft with-helm=true 
- [x] /werft with-clean-slate-deployment=

File: dev/image/Dockerfile
Comment (kylos101) [inline dev/image/Dockerfile:36]: Hey @princerachit can you try removing these, and rebuilding the dev image with the latest `europe-docker.pkg.dev/gitpod-artifacts/docker-dev/workspace-base-images:full`? It should work better no, and no longer get a NO_PUBKEY error.

File: dev/image/Dockerfile
Comment (kylos101) [inline dev/image/Dockerfile:148]: Can you share a link to the build in werft that necessitated this change? We can begin dog fooding like this, but ideally this change wouldn't be necessary.

My thought is we should be able to swap like for like when going from the current full on docker hub to this new one. At least, if it's possible.

Comment (princerachit) [reply dev/image/Dockerfile:148]: One example https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.70

Comment (kylos101) [reply dev/image/Dockerfile:145]: Got it, thank you! 

The error was permissions related:
```shell
[dev/image:docker] Step 38/63 : RUN bash -c "python3 -m pip uninstall crcmod; python3 -m pip install --no-cache-dir -U crcmod"
[dev/image:docker]  ---> Running in 6f7a4af96876
[dev/image:docker] [91mWARNING: Skipping crcmod as it is not installed.
[dev/image:docker] [0mCollecting crcmod
[dev/image:docker]   Downloading crcmod-1.7.tar.gz (89 kB)
[dev/image:docker] Building wheels for collected packages: crcmod
[dev/image:docker]   Building wheel for crcmod (setup.py): started
[dev/image:docker]   Building wheel for crcmod (setup.py): finished with status 'done'
[dev/image:docker]   Created wheel for crcmod: filename=crcmod-1.7-py3-none-any.whl size=18840 sha256=580e9c7e56175ac26f60175c7b1592cdbac0007696934f9cc4b5189b1fe9fdd7
[dev/image:docker]   Stored in directory: /tmp/pip-ephem-wheel-cache-zlxkbes3/wheels/ca/5a/02/f3acf982a026f3319fb3e798a8dca2d48fafee7761788562e9
[dev/image:docker] Successfully built crcmod
[dev/image:docker] Installing collected packages: crcmod
[dev/image:docker] [91mERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/workspace/.pip-modules'
[dev/image:docker] Check the permissions.
```

Let's plan on solving in a follow-up issue, while the org is dog fooding.

File: dev/image/Dockerfile
Comment (kylos101) [inline dev/image/Dockerfile:8]: We need to explicitly test the leeway cache, running a full build of Gitpod, without any prior cache, to assert that we're not missing anything.

`-c` flag should help with this, refer to the build.ts as an example to see current usage (which relies on cache).

Comment (kylos101) [reply dev/image/Dockerfile:8]: Hey @princerachit in talking with @csweichel he recommended that we run a build with caching disabled in `Leeway`. 

To do this I:
1. patched `./.werft/build.ts` in the five spots where `leeway build` is called to include `-c none`. Four spots were using `-c remote`, one did not specify `c` or `caching` at all.
2. sideloaded the updated build.ts (using no cache) file as part of the werft job like so `werft run github -j ./.werft/build.yaml -s ./.werft/build.ts `

Good news, and bad news! 

It appears to have caught an issue, which requires research.

[This build timed out](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.86), can you check to see why, and if the test is still valid? I suspect the test is starting too soon, and may need its timeout increased, or to be delayed somehow.

We'll have to resolve this prior to accepting the PR and beginning dog-fooding.

Comment (princerachit) [reply dev/image/Dockerfile:8]: I tried rerunning the tests. Build [failed](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.90/logs#build:components/ide/code:docker) due to nodejs/yarn error and then it [succeeded](https://werft.gitpod-dev.com/job/gitpod-build-prs-test-dazzle-rewrite.91) on rerun. The issues doesn't seem to be related to the PR. I have reached out to the webapp team to seek more information.

Comment (kylos101) [reply dev/image/Dockerfile:8]: Nice! I see you increased the timeout too. Thank yous.

**Comment by kylos101** [comment]
@aledbf can we get a review from you?

note: we consulted this PR to see which files must be updated when the dev image is changed, https://github.com/gitpod-io/gitpod/pull/7387/files

**Comment by kylos101** [comment]
Setting to draft, until core-dev stabilizes.

**Comment by kylos101** [comment]
Hey @aledbf I just moved this from draft back to ready for review. Considering some folks are able to continue working in `core-dev`, it would be good to get this out there, so that we can begin dog fooding to surface any potential issues.

**Review by kylos101 — APPROVED** [review]
/approve

Testing so far has gone smooth. At this point we require feedback from Gitpodders.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 76655d3d7a55ffebd34bb47cd30d2b273a6c87a3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7595#pullrequestreview-856900183" title="Approved">kylos101</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **princerachit** after the PR has been reviewed.

Associated issue: *#7042*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7595#issuecomment-1016637054" title="Approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7595#pullrequestreview-856900183" title="Approved">kylos101</a>*

Associated issue: *#7042*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
